src/test/java/com/datalogics/pdf/samples/signature/SignDocumentTest.java

```diff
@@ -43,64 +52,101 @@
  * Tests the SignDocument Sample.
  */
 public class SignDocumentTest extends SampleTest {
-    static final String FILE_NAME = "SignedField.pdf";
-    static final String QUALIFIED_SIGNATURE_FIELD_NAME = "Approver";
+    private static final String FILE_NAME = "SignedField.pdf";
+    private static final String QUALIFIED_SIGNATURE_FIELD_NAME = "Approver";
     private static final String BLACK_RIGHTWARDS_ARROW = "\u27a1"; // U+27A1 BLACK RIGHTWARDS ARROW
+    private static File file = null;
+    private static PDFDocument pdfDocument;
 
-    @Test
-    public void testSignExistingSignatureFields() throws Exception {
-        final URL inputUrl = SignDocument.class.getResource(SignDocument.INPUT_UNSIGNED_PDF_PATH);
+    @BeforeClass
+    public static void setUpDocument() throws Exception {
+        pdfDocument = null;
+    }
 
-        final File file = SampleTest.newOutputFileWithDelete(FILE_NAME);
+    /**
+     * Create the document if this is the first test run in the class.
+     *
+     * @throws Exception a general exception was thrown
+     */
+    private static void ensureDocument() throws Exception {
```

Comment:
I think you could call ensureDocument in setUpDocument. Then you can avoid calling ensureDocument 3 times for each test. They don't seem to be manipulating the document so it should be okay.